### PR TITLE
Send alerts to slack in smaug

### PR DESCRIPTION
**Pull request by HumairAK**
Resolve: https://github.com/operate-first/apps/issues/1975

Alertmanager config requires you to put the slack api url within the
alertmanager secret. This is unfortunate because we would like to keep
most of the alertmanager secret visible to the public as all our routing
and inhibit rules go there.

To get around this, we build the alert manager secret using ESO by
pulling from the alert-manager-template configmap. ESO will patch in the
slack api url from vault upon deployment and create the alertmanager-main
secret in openshift-monitoring.

To better accomodate this change the commit restructures configmaps &
secrets within the smaug overlay so that they are confined to their
respective sub folders.

Related: 
- https://github.com/operate-first/support/issues/298
- https://github.com/operate-first/operations/issues/468

### CODE REVIEW

File: cluster-scope/overlays/prod/moc/smaug/configmaps/files/alertmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        - severity = critical
+      group_by:
+        - service
+      receiver: slack
+    - matchers:
+        - namespace = openshift-storage
+        - alertname = KubePersistentVolumeFillingUp
+        - severity = critical
+      receiver: slack
+      group_interval: 12h
```
Comment (HumairAK) [inline cluster-scope/overlays/prod/moc/smaug/configmaps/files/alertmanager.yaml:80]: @larsks starting off with these slack alerts based on our slack discussions last week, wdyt?

**Comment by HumairAK** [comment]
We will also need to propogate this set up to the alertmanager secrets being managed in the other clusters.

File: cluster-scope/overlays/prod/moc/smaug/secrets/kustomization.yaml [hunk 2]
```diff
@@ -2,10 +2,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 generatorOptions:
   disableNameSuffixHash: true
-secretGenerator:
-  - files:
-      - files/alertmanager.yaml
-    name: alertmanager-main
-    namespace: openshift-monitoring
 generators:
   - secret-generator.yaml
```
Comment (Gregory-Pereira) [inline cluster-scope/overlays/prod/moc/smaug/secrets/kustomization.yaml:6]: Shouldnt this be removed as well?

**Review by Gregory-Pereira — COMMENTED** [review]
/lgtm
Just one question ...

File: cluster-scope/overlays/prod/moc/smaug/secrets/kustomization.yaml (hunk 2 above)
Comment (HumairAK) [reply cluster-scope/overlays/prod/moc/smaug/secrets/kustomization.yaml:6]: why?

**Pull request merged**